src/DigiTicketsVoucherGateway.php

```diff
@@ -72,13 +76,31 @@ public function unredeem(array $parameters = array())
 
     public function setVirtualApi($value)
     {
-error_log('[Driver] setVirtualApi is being called');
         $this->setParameter('virtualApi', $value);
     }
 
     public function getVirtualApi()
     {
-error_log('[Driver] getVirtualApi is being called');
         return $this->getParameter('virtualApi');
     }
-}
\ No newline at end of file
+
+    /**
+     * Pass in a callback, with properties of type (string $message, array $data=null)
+     * @param callable $callback
+     *
+     * @return void
+     */
+    public function registerLoggerCallback(callable $callback)
+    {
+        $this->logger = $callback;
+        $this->uniqueLogKey = uniqid();
+    }
+
+    public function log(string $message, array $data = null)
+    {
+        if (is_callable($this->logger)) {
+            $message = '[DTVoucher-'.$this->uniqueLogKey.'] '.$message;
+            ($this->logger)($message, $data);
```

Comment:
@jamesgarton I guess it makes more sense to call this property a callback not logger, as it is a logger callback, but it does not really matter